frontend/src/utils/course-helpers.ts

```diff
@@ -55,3 +57,32 @@ export async function getScheduleCourseCoreqs(
   }
   return coursesCoreqs;
 }
+
+/**
+ * Flattens the Requirement[] into only a list of Requirements/Requirement sets
+ * This means that all inner lists will only contain one class or a list of the primary class and its labs/recitations
+ * @param reqs
+ */
+export function flatten(reqs: Requirement[]): IRequiredCourse[][] {
+  return reqs.map(flattenOne).reduce((array, cur) => array.concat(cur), []);
+}
+
+function flattenOne(req: Requirement): IRequiredCourse[][] {
+  if (req.type === "COURSE") {
+    return [[req as IRequiredCourse]];
+  } else if (
+    req.type === "AND" &&
+    req.courses.filter(c => c.type === "COURSE").length
+  ) {
+    return [req.courses as IRequiredCourse[]];
+  } else if (req.type === "AND" || req.type === "OR") {
+    return flatten(req.courses);
+  } else {
+    return [];
+  }
+}
+
+export const courseToString = (c: {
+  subject: string;
+  classId: number | string;
+}) => `${c.subject}${c.classId}`;
```

Comment:
Thank you!!! I think we have course to string being done in so many places but nobody bothered to standardize it. If you have time and saw some of it would really appreciate it if you could add this function but don't spend more than a few minutes!(hint: line 114 in CompletedCoursesScreen)

Reply:
i'll do a look over!

Reply:
done